### PR TITLE
Better number format for memory usage

### PR DESCRIPTION
**Pull request by volzhs**
before
screenshot from 2017-11-24 22-53-51

after
screenshot from 2017-11-24 22-52-50

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Review by bojidar-bg — APPROVED** [review]

File: editor/script_editor_debugger.cpp [hunk 1]
```diff
@@ -642,7 +642,7 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, const Array &p_da
 							v /= 1024.0;
 						}
 						tt += " bytes";
-						vs = rtos(v) + " " + unit;
+						vs = rtos(Math::round(v * 100) * 0.01) + " " + unit;
```
Comment (akien-mga) [inline editor/script_editor_debugger.cpp]: I'm pretty sure we have a built-in and more reliable way to do this. You coudl still get `32.00001 MB` with this method. There should be simple C float formatting instructions available.

Comment (bojidar-bg) [reply editor/script_editor_debugger.cpp]: There is `static String::num()`, which takes in the number of decimals, and there is `static String::sprintf()`

Comment (akien-mga) [reply editor/script_editor_debugger.cpp]: Well maybe I spoke too fast, I don't find a simpler way to do this. The alternative would be snprintf with two buffers...

Comment (akien-mga) [reply editor/script_editor_debugger.cpp]: Ah of course in String yeah, I was looking at Math...

Comment (volzhs) [reply editor/script_editor_debugger.cpp]: so... what would be the best for it?

Comment (akien-mga) [reply editor/script_editor_debugger.cpp]: `String::num(v, 2)`

Comment (volzhs) [reply editor/script_editor_debugger.cpp]: I couldn't find this beautiful method on GDscript.
why didn't this bind to gdscript?

Comment (vnen) [reply editor/script_editor_debugger.cpp]: @volzhs there's [`String.pad_decimals()`](http://docs.godotengine.org/en/latest/classes/class_string.html#class-string-pad-decimals) in GDScript, but you can also use [format string](http://docs.godotengine.org/en/latest/learning/scripting/gdscript/gdscript_format_string.html) to achieve this.

Comment (volzhs) [reply editor/script_editor_debugger.cpp]: @vnen Thanks for the tip. I'm using format string. but I'm not aware of `pad_decimals`

File: editor/script_editor_debugger.cpp [hunk 2]
```diff
@@ -642,7 +643,7 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, const Array &p_da
 							v /= 1024.0;
 						}
 						tt += " bytes";
-						vs = rtos(v) + " " + unit;
+						vs = String::num(v, 2) + " " + unit;
 					} break;
 					case Performance::MONITOR_TYPE_TIME: {
 						tt += " seconds";
```
Comment (djrm) [inline editor/script_editor_debugger.cpp]: ugh, i forgot to TTR this.

Comment (volzhs) [reply editor/script_editor_debugger.cpp]: I thought about that also. but I doubted if it's necessary.

**Pull request merged**